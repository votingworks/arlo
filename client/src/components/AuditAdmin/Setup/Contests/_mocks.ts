export const contestsInputMocks = {
  inputs: [
    { key: 'Contest Name', value: 'Contest Name' },
    { key: 'Name of Candidate/Choice 1', value: 'Choice One' },
    { key: 'Name of Candidate/Choice 2', value: 'Choice Two' },
    { key: 'Votes for Candidate/Choice 1', value: '10' },
    { key: 'Votes for Candidate/Choice 2', value: '20' },
    { key: 'Total Ballot Cards Cast for Contest', value: '30' },
  ],
  errorInputs: [
    { key: 'Contest Name', value: '', error: 'Required' },
    {
      key: 'Total Ballot Cards Cast for Contest',
      value: '',
      error:
        'Must be greater than or equal to the sum of votes for each candidate/choice',
    },
    {
      key: 'Total Ballot Cards Cast for Contest',
      value: 'test',
      error: 'Must be a number',
    },
    {
      key: 'Total Ballot Cards Cast for Contest',
      value: '-1',
      error: 'Must be a positive number',
    },
    {
      key: 'Total Ballot Cards Cast for Contest',
      value: '0.5',
      error: 'Must be an integer',
    },
    { key: 'Name of Candidate/Choice 1', value: '', error: 'Required' },
    { key: 'Name of Candidate/Choice 2', value: '', error: 'Required' },
    {
      key: 'Votes for Candidate/Choice 1',
      value: '',
      error: 'Required',
    },
    {
      key: 'Votes for Candidate/Choice 1',
      value: 'test',
      error: 'Must be a number',
    },
    {
      key: 'Votes for Candidate/Choice 1',
      value: '-1',
      error: 'Must be a positive number',
    },
    {
      key: 'Votes for Candidate/Choice 1',
      value: '0.5',
      error: 'Must be an integer',
    },
    {
      key: 'Votes for Candidate/Choice 2',
      value: '',
      error: 'Required',
    },
    {
      key: 'Votes for Candidate/Choice 2',
      value: 'test',
      error: 'Must be a number',
    },
    {
      key: 'Votes for Candidate/Choice 2',
      value: '-1',
      error: 'Must be a positive number',
    },
    {
      key: 'Votes for Candidate/Choice 2',
      value: '0.5',
      error: 'Must be an integer',
    },
  ],
}

export default contestsInputMocks
