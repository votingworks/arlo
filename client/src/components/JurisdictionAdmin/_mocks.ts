import { IBatches } from './useBatchResults'
import { IRound } from '../AuditAdmin/useRoundsAuditAdmin'
import { FileProcessingStatus } from '../useCSV'
import {
  IFullHandTallyBatchResults,
  IFullHandTallyBatchResult,
} from './useFullHandTallyResults'
import { mocksOfType } from '../testUtilities'

export interface INullResultValues {
  [contestId: string]: {
    [choiceId: string]: null | string | number
  }
}

export const roundMocks = mocksOfType<IRound>()({
  incomplete: {
    id: 'round-1',
    roundNum: 1,
    startedAt: '2020-09-14T17:35:19.482Z',
    endedAt: null,
    isAuditComplete: false,
    needsFullHandTally: false,
    isFullHandTally: false,
    drawSampleTask: {
      status: FileProcessingStatus.PROCESSED,
      startedAt: '2020-09-14T17:35:19.482Z',
      completedAt: '2020-09-14T17:36:19.482Z',
      error: null,
    },
  },
  complete: {
    id: 'round-1',
    roundNum: 1,
    startedAt: '2020-09-14T17:35:19.482Z',
    endedAt: '2020-09-14T17:35:19.482Z',
    needsFullHandTally: false,
    isAuditComplete: true,
    isFullHandTally: false,
    drawSampleTask: {
      status: FileProcessingStatus.PROCESSED,
      startedAt: '2020-09-14T17:35:19.482Z',
      completedAt: '2020-09-14T17:36:19.482Z',
      error: null,
    },
  },
  fullHandTallyIncomplete: {
    id: 'round-1',
    roundNum: 1,
    startedAt: '2020-09-14T17:35:19.482Z',
    endedAt: null,
    isAuditComplete: false,
    needsFullHandTally: true,
    isFullHandTally: true,
    drawSampleTask: {
      status: FileProcessingStatus.PROCESSED,
      startedAt: '2020-09-14T17:35:19.482Z',
      completedAt: '2020-09-14T17:36:19.482Z',
      error: null,
    },
  },
})

export const resultsMocks = mocksOfType<INullResultValues>()({
  emptyInitial: {
    'contest-id-1': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
    'contest-id-2': {
      'choice-id-3': null,
      'choice-id-4': null,
    },
  },
  complete: {
    'contest-id-1': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
    'contest-id-2': {
      'choice-id-3': 1,
      'choice-id-4': 2,
    },
  },
})

export const batchResultsMocks = mocksOfType<INullResultValues>()({
  empty: {
    'batch-1': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
    'batch-2': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
    'batch-3': {
      'choice-id-1': null,
      'choice-id-2': null,
    },
  },
  complete: {
    'batch-1': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
    'batch-2': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
    'batch-3': {
      'choice-id-1': 1,
      'choice-id-2': 2,
    },
  },
})

export const batchesMocks = mocksOfType<IBatches>()({
  emptyInitial: {
    batches: [
      {
        id: 'batch-1',
        name: 'Batch One',
        numBallots: 100,
        auditBoard: null,
        resultTallySheets: [],
      },
      {
        id: 'batch-2',
        name: 'Batch Two',
        numBallots: 100,
        auditBoard: null,
        resultTallySheets: [],
      },
      {
        id: 'batch-3',
        name: 'Batch Three',
        numBallots: 100,
        auditBoard: null,
        resultTallySheets: [],
      },
    ],
    resultsFinalizedAt: null,
  },
  complete: {
    batches: [
      {
        id: 'batch-1',
        name: 'Batch One',
        numBallots: 100,
        auditBoard: {
          id: 'ab-1',
          name: 'Audit Board One',
        },
        resultTallySheets: [
          {
            name: 'Tally Sheet #1',
            results: {
              'choice-id-1': 1,
              'choice-id-2': 2,
            },
          },
        ],
      },
      {
        id: 'batch-2',
        name: 'Batch Two',
        numBallots: 100,
        auditBoard: {
          id: 'ab-1',
          name: 'Audit Board One',
        },
        resultTallySheets: [
          {
            name: 'Tally Sheet #1',
            results: {
              'choice-id-1': 0,
              'choice-id-2': 10,
            },
          },
        ],
      },
      {
        id: 'batch-3',
        name: 'Batch Three',
        numBallots: 100,
        auditBoard: {
          id: 'ab-1',
          name: 'Audit Board One',
        },
        resultTallySheets: [
          {
            name: 'Tally Sheet #1',
            results: {
              'choice-id-1': 2000,
              'choice-id-2': 20,
            },
          },
        ],
      },
    ],
    resultsFinalizedAt: '2020-09-14T17:35:19.482Z',
  },
})

export const fullHandTallyBatchResultMock = mocksOfType<
  IFullHandTallyBatchResults
>()({
  empty: {
    finalizedAt: '',
    results: [],
  },
  complete: {
    finalizedAt: '',
    results: [
      {
        batchName: 'Batch1',
        batchType: 'Other',
        choiceResults: {
          'choice-id-1': 10,
          'choice-id-2': 20,
        },
      },
    ],
  },
  completeWithMultipleBatch: {
    finalizedAt: '',
    results: [
      {
        batchName: 'Batch1',
        batchType: 'Other',
        choiceResults: {
          'choice-id-1': 5,
          'choice-id-2': 15,
        },
      },
      {
        batchName: 'Batch2',
        batchType: 'Provisional',
        choiceResults: {
          'choice-id-1': 5,
          'choice-id-2': 5,
        },
      },
    ],
  },
  updated: {
    finalizedAt: '',
    results: [
      {
        batchName: 'Batch12',
        batchType: 'Other',
        choiceResults: {
          'choice-id-1': 10,
          'choice-id-2': 20,
        },
      },
    ],
  },
  finalized: {
    finalizedAt: '2021-04-13T15:01:00.383031+00:00',
    results: [
      {
        batchName: 'Batch12',
        batchType: 'Other',
        choiceResults: {
          'choice-id-1': 10,
          'choice-id-2': 20,
        },
      },
    ],
  },
})

export const fullHandTallyBatchResultsMock = mocksOfType<
  IFullHandTallyBatchResult
>()({
  empty: {
    batchName: '',
    batchType: '',
    choiceResults: {},
  },
  complete: {
    batchName: 'Batch1',
    batchType: 'Other',
    choiceResults: {
      'choice-id-1': 10,
      'choice-id-2': 20,
    },
  },
  updated: {
    batchName: 'Batch12',
    batchType: 'Other',
    choiceResults: {
      'choice-id-1': 10,
      'choice-id-2': 20,
    },
  },
})
