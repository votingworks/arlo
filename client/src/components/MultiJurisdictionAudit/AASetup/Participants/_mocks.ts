import { readFileSync } from 'fs'
import { join } from 'path'

const jurisdictionFile = new File(
  [
    readFileSync(
      join(
        __dirname,
        '../../../../../public/sample_jurisdiction_filesheet.csv'
      ),
      'utf8'
    ),
  ],
  'jurisdictions.csv',
  { type: 'text/csv' }
)

export default jurisdictionFile
