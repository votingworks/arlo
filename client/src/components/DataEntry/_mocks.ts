import { IAuditBoard, BallotStatus, Interpretation } from '../../types'
import { contestMocks } from '../MultiJurisdictionAudit/Setup/Contests/_mocks'
import { IBallot } from './Ballot'

export const contest = contestMocks.filledTargeted.contests[0]

export const doneDummyBallots: { ballots: IBallot[] } = {
  ballots: [
    {
      id: 'ballot-id-1',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-2',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-3',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-4',
      batch: {
        name: '0003-04-Precinct 31 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 1965,
      status: BallotStatus.NOT_FOUND,
      interpretations: [],
    },
  ],
}

export const dummyBallots: { ballots: IBallot[] } = {
  ballots: [
    {
      id: 'ballot-id-1',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: null,
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [
        {
          contestId: contest.id,
          interpretation: Interpretation.VOTE,
          choiceIds: [contest.choices[0].id],
          comment: 'Good ballot',
        },
      ],
    },
    {
      id: 'ballot-id-2',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-3',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-1',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-4',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-2',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-5',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-2',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-6',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-2',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-7',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-3',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-8',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-3',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-9',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-3',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-10',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-4',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-11',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-4',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-12',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-4',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-13',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-5',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-14',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-5',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-15',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-5',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-16',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-6',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-17',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-6',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-18',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-6',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-19',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-7',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-20',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-7',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-21',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-7',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-22',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-8',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-23',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-8',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-24',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-8',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-25',
      batch: {
        name: '0003-04-Precinct 13 (Jonesboro Fire Department)',
        id: 'batch-id-9',
        tabulator: '11',
      },
      position: 313,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-26',
      batch: {
        name: '0003-04-Precinct 19 (Jonesboro Fire Department)',
        id: 'batch-id-9',
        tabulator: '11',
      },
      position: 2112,
      status: BallotStatus.NOT_AUDITED,
      interpretations: [],
    },
    {
      id: 'ballot-id-27',
      batch: {
        name: '0003-04-Precinct 29 (Jonesboro Fire Department)',
        id: 'batch-id-9',
        tabulator: '11',
      },
      position: 1789,
      status: BallotStatus.AUDITED,
      interpretations: [],
    },
  ],
}

export const dummyBoards = (): IAuditBoard[] => [
  {
    id: 'audit-board-1',
    name: 'Audit Board #1',
    jurisdictionId: 'jurisdiction-1',
    jurisdictionName: 'Jurisdiction 1',
    roundId: 'round-1',
    members: [
      {
        name: 'John Doe',
        affiliation: '',
      },
      {
        name: 'Jane Doe',
        affiliation: 'LIB',
      },
    ],
    signedOffAt: null,
  },
  {
    id: 'audit-board-2',
    name: 'Audit Board #2',
    jurisdictionId: 'jurisdiction-1',
    jurisdictionName: 'Jurisdiction 1',
    roundId: 'round-1',
    members: [],
    signedOffAt: null,
  },
]

export default {
  dummyBoards,
  dummyBallots,
}
