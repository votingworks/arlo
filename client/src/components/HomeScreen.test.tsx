import React from 'react'
import { screen, within } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { withMockFetch, renderWithRouter } from './testUtilities'
import App from '../App'
import { aaApiCalls, jaApiCalls } from './MultiJurisdictionAudit/_mocks'
import { auditSettings } from './MultiJurisdictionAudit/useSetupMenuItems/_mocks'

const apiCalls = {
  unauthenticatedUser: {
    url: '/api/me',
    response: null,
  },
  postNewAudit: (body: {}) => ({
    url: '/api/election',
    options: {
      method: 'POST',
      body: JSON.stringify(body),
      headers: {
        'Content-Type': 'application/json',
      },
    },
    response: { electionId: '1' },
  }),
  getUserWithAudit: {
    ...aaApiCalls.getUser,
    response: {
      ...aaApiCalls.getUser.response,
      organizations: [
        {
          id: 'org-id',
          name: 'State of California',
          elections: [
            {
              id: '1',
              auditName: 'November Presidential Election 2020',
              electionName: '',
              state: 'CA',
            },
          ],
        },
      ],
    },
  },
  getUserMultipleOrgs: {
    ...aaApiCalls.getUser,
    response: {
      ...aaApiCalls.getUser.response,
      organizations: [
        {
          id: 'org-id',
          name: 'State of California',
          elections: [
            {
              id: '1',
              auditName: 'November Presidential Election 2020',
              electionName: '',
              state: 'CA',
            },
          ],
        },
        {
          id: 'org-id-2',
          name: 'State of Georgia',
          elections: [],
        },
      ],
    },
  },
}

const setupScreenCalls = [
  aaApiCalls.getRounds,
  aaApiCalls.getJurisdictions,
  aaApiCalls.getContests,
  aaApiCalls.getSettings(auditSettings.blank),
]

const renderView = (route: string) => renderWithRouter(<App />, { route })

describe('Home screen', () => {
  it('shows a login screen for unauthenticated users', async () => {
    const expectedCalls = [apiCalls.unauthenticatedUser]
    await withMockFetch(expectedCalls, async () => {
      renderView('/')
      await screen.findByRole('img', { name: 'Arlo, by VotingWorks' })
      const jaLoginButton = screen.getByRole('button', {
        name: 'Log in to your audit',
      })
      expect(jaLoginButton).toHaveAttribute(
        'href',
        '/auth/jurisdictionadmin/start'
      )
      const aaLoginButton = screen.getByRole('link', {
        name: 'Log in as an admin',
      })
      expect(aaLoginButton).toHaveAttribute('href', '/auth/auditadmin/start')
    })
  })

  it('shows a message when logged out for inactivity', async () => {
    const expectedCalls = [apiCalls.unauthenticatedUser]
    await withMockFetch(expectedCalls, async () => {
      renderView('/#logged-out')
      await screen.findByText('You have been logged out due to inactivity.')
    })
  })

  it.skip('shows a list of audits and create audit form for audit admins', async () => {
    // TEST TODO
    const expectedCalls = [
      aaApiCalls.getUser,
      aaApiCalls.getUser, // Extra call to load the list of audits
      apiCalls.postNewAudit({
        organizationId: 'org-id',
        auditName: 'November Presidential Election 2020',
        auditType: 'BATCH_COMPARISON',
        auditMathType: 'MACRO',
      }),
      ...setupScreenCalls,
      aaApiCalls.getJurisdictionFile,
      aaApiCalls.getRounds,
      ...setupScreenCalls,
      aaApiCalls.getSettings(auditSettings.blank),
      aaApiCalls.getJurisdictionFile,
      apiCalls.getUserWithAudit,
      ...setupScreenCalls,
      aaApiCalls.getJurisdictionFile,
      aaApiCalls.getRounds,
      ...setupScreenCalls,
      aaApiCalls.getSettings(auditSettings.blank),
      aaApiCalls.getJurisdictionFile,
    ]
    await withMockFetch(expectedCalls, async () => {
      const { history } = renderView('/')
      await screen.findByRole('heading', {
        name: 'Audits - State of California',
      })
      screen.getByText(
        "You haven't created any audits yet for State of California"
      )

      // Try to create an audit without typing in an audit name
      screen.getByRole('heading', { name: 'New Audit' })
      const createAuditButton = screen.getByRole('button', {
        name: 'Create Audit',
      })
      userEvent.click(createAuditButton)
      const auditNameInput = screen.getByRole('textbox', { name: 'Audit name' })
      await within(auditNameInput.closest('label')!).findByText('Required')

      // Create a new audit
      await userEvent.type(
        auditNameInput,
        'November Presidential Election 2020'
      )
      expect(
        screen.getByRole('radio', { name: 'Ballot Polling' })
      ).toBeChecked()
      userEvent.click(screen.getByRole('radio', { name: 'Batch Comparison' }))
      userEvent.click(createAuditButton)

      // Should be on the setup screen
      await screen.findByText('The audit has not started.')
      expect(history.location.pathname).toEqual('/election/1/setup')

      // Go back to the home screen
      userEvent.click(screen.getByRole('link', { name: 'View Audits' }))

      // Click on the audit to go the setup screen
      userEvent.click(
        await screen.findByRole('button', {
          name: 'November Presidential Election 2020',
        })
      )
      await screen.findByText('The audit has not started.')
      expect(history.location.pathname).toEqual('/election/1/setup')
    })
  })

  it.skip('shows a list of audits and create audit form for audit admins with multiple orgs', async () => {
    // TEST TODO
    const expectedCalls = [
      apiCalls.getUserMultipleOrgs,
      apiCalls.getUserMultipleOrgs,
      apiCalls.postNewAudit({
        organizationId: 'org-id-2',
        auditName: 'Presidential Primary',
        auditType: 'BALLOT_POLLING',
        auditMathType: 'BRAVO',
      }),
      ...setupScreenCalls,
      aaApiCalls.getJurisdictionFile,
      aaApiCalls.getRounds,
      ...setupScreenCalls,
      aaApiCalls.getSettings(auditSettings.blank),
      aaApiCalls.getJurisdictionFile,
    ]
    await withMockFetch(expectedCalls, async () => {
      renderView('/')

      // Two orgs and their audits get displayed
      const californiaHeading = await screen.findByRole('heading', {
        name: 'Audits - State of California',
      })
      within(californiaHeading.closest('div')!).getByRole('button', {
        name: 'November Presidential Election 2020',
      })
      const georgiaHeading = screen.getByRole('heading', {
        name: 'Audits - State of Georgia',
      })
      within(georgiaHeading.closest('div')!).getByText(
        "You haven't created any audits yet for State of Georgia"
      )

      // Select an organization
      const orgSelect = screen.getByRole('combobox', { name: /Organization/ })
      expect(
        screen.getByRole('option', {
          name: 'State of California',
        })
      ).toHaveProperty('selected', true)
      userEvent.selectOptions(orgSelect, [
        screen.getByRole('option', {
          name: 'State of Georgia',
        }),
      ])

      // Create a new audit
      await userEvent.type(
        screen.getByRole('textbox', { name: 'Audit name' }),
        'Presidential Primary'
      )
      userEvent.click(
        screen.getByRole('button', {
          name: 'Create Audit',
        })
      )

      // Should be on the setup screen
      await screen.findByText('The audit has not started.')
    })
  })

  it.skip('shows a list of audits for jurisdiction admins', async () => {
    // TEST TODO
    const expectedCalls = [
      jaApiCalls.getUser,
      jaApiCalls.getSettings(auditSettings.blank),
      jaApiCalls.getRounds,
      jaApiCalls.getBallotManifestFile({ file: null, processing: null }),
      jaApiCalls.getBatchTalliesFile({ file: null, processing: null }),
    ]
    await withMockFetch(expectedCalls, async () => {
      renderView('/')

      // Two audits and their jurisdictions get displayed
      const auditOneHeading = await screen.findByRole('heading', {
        name: 'Jurisdictions - audit one',
      })
      const j1Button = within(auditOneHeading.closest('div')!).getByRole(
        'button',
        {
          name: 'Jurisdiction One',
        }
      )
      within(auditOneHeading.closest('div')!).getByRole('button', {
        name: 'Jurisdiction Three',
      })
      const auditTwoHeading = await screen.findByRole('heading', {
        name: 'Jurisdictions - audit two',
      })
      within(auditTwoHeading.closest('div')!).getByRole('button', {
        name: 'Jurisdiction Two',
      })

      // Click on a jurisdiction to go to the audit
      userEvent.click(j1Button)
      await screen.findByText('The audit has not started.')
    })
  })
})
