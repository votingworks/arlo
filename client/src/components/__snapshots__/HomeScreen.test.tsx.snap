// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home screen redirects to audit screen if only one election exists for JA 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="sc-exAgwC gHrCj"
  >
    <div
      class="bp3-navbar sc-bxivhb gvSyoM"
    >
      <div
        class="sc-bwzfXH sc-bZQynM ezCmGO"
      >
        <div
          class="bp3-navbar-group bp3-align-left"
        >
          <div
            class="bp3-navbar-heading"
          >
            <a
              href="/"
            >
              <img
                alt="Arlo, by VotingWorks"
                src="/arlo.png"
              />
            </a>
          </div>
          <div
            class="bp3-navbar-heading"
          >
            Jurisdiction: 
            Jurisdiction One
          </div>
        </div>
        <div
          class="bp3-navbar-group bp3-align-right"
        >
          <div
            class="sc-EHOje fOUAJN"
          >
            <div
              class="bp3-popover-wrapper bp3-fill"
            >
              <div
                class="bp3-popover-target"
              >
                <button
                  class="bp3-button bp3-minimal"
                  type="button"
                >
                  <span
                    class="bp3-icon bp3-icon-user"
                    icon="user"
                  >
                    <svg
                      data-icon="user"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <desc>
                        user
                      </desc>
                      <path
                        d="M7.99-.01A7.998 7.998 0 00.03 8.77c.01.09.03.18.04.28.02.15.04.31.07.47.02.11.05.22.08.34.03.13.06.26.1.38.04.12.08.25.12.37.04.11.08.21.12.32a6.583 6.583 0 00.3.65c.07.14.14.27.22.4.04.07.08.13.12.2l.27.42.1.13a7.973 7.973 0 003.83 2.82c.03.01.05.02.07.03.37.12.75.22 1.14.29l.2.03c.39.06.79.1 1.2.1s.81-.04 1.2-.1l.2-.03c.39-.07.77-.16 1.14-.29.03-.01.05-.02.07-.03a8.037 8.037 0 003.83-2.82c.03-.04.06-.08.09-.13.1-.14.19-.28.28-.42.04-.07.08-.13.12-.2.08-.13.15-.27.22-.41.04-.08.08-.17.12-.26.06-.13.11-.26.17-.39.04-.1.08-.21.12-.32.04-.12.08-.24.12-.37.04-.13.07-.25.1-.38.03-.11.06-.22.08-.34.03-.16.05-.31.07-.47.01-.09.03-.18.04-.28.02-.26.04-.51.04-.78-.03-4.41-3.61-7.99-8.03-7.99zm0 14.4c-1.98 0-3.75-.9-4.92-2.31.67-.36 1.49-.66 2.14-.95 1.16-.52 1.04-.84 1.08-1.27.01-.06.01-.11.01-.17-.41-.36-.74-.86-.96-1.44v-.01c0-.01-.01-.02-.01-.02-.05-.13-.09-.26-.12-.39-.28-.05-.44-.35-.5-.63-.06-.11-.18-.38-.15-.69.04-.41.2-.59.38-.67v-.06c0-.51.05-1.24.14-1.72.02-.13.05-.26.09-.39.17-.59.53-1.12 1.01-1.49.49-.38 1.19-.59 1.82-.59.62 0 1.32.2 1.82.59.48.37.84.9 1.01 1.49.04.13.07.26.09.4.09.48.14 1.21.14 1.72v.07c.18.08.33.26.37.66.03.31-.1.58-.16.69-.06.27-.21.57-.48.62-.03.13-.07.26-.12.38 0 .01-.01.04-.01.04-.21.57-.54 1.06-.94 1.42 0 .06.01.13.01.19.04.43-.12.75 1.05 1.27.65.29 1.47.6 2.14.95a6.415 6.415 0 01-4.93 2.31z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                  <span
                    class="bp3-button-text"
                  >
                    jurisdictionadmin@email.org
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-bdVaJa bKocXA"
    >
      <div
        class="bp3-callout sc-gPEVay bivkYG"
      >
        <div
          class="sc-bwzfXH jiphYj"
        >
          <div
            class="text"
          >
            <h3
              class="bp3-heading sc-jDwBTQ kSuPTu"
            >
              Test Audit
            </h3>
            <h4
              class="bp3-heading"
            >
              The audit has not started.
            </h4>
            <p>
              Ballot manifest not uploaded.
            </p>
          </div>
        </div>
      </div>
      <div
        class="sc-bwzfXH sc-hmzhuo hwfNhd"
      >
        <h2
          class="bp3-heading sc-dnqmqq dMLSzf"
        >
          Audit Source Data
        </h2>
        <form>
          <div
            class="sc-fjdhpX jaSwrG"
          >
            <h2
              class="bp3-heading sc-dnqmqq dMLSzf"
            />
            <div
              class="sc-kAzzGY iPDebS"
            >
              <h4
                class="bp3-heading"
              >
                Ballot Manifest
              </h4>
              <div
                class="sc-chPdSV dGLcDt"
              >
                Click "Browse" to choose the appropriate Ballot
                Manifest file from your computer. This file should be a
                comma-separated list of all the ballot boxes/containers used
                to store ballots for this particular election, plus a count of
                how many ballot cards (individual pieces of paper) are stored
                in each container.
                <br />
                <br />
                <a
                  href="/sample_ballot_manifest.csv"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  (Click here to view a sample file in the correct format.)
                </a>
              </div>
            </div>
            <div
              class="sc-kAzzGY iPDebS"
            >
              <label
                class="bp3-file-input"
              >
                <input
                  accept=".csv"
                  name="csv"
                  type="file"
                />
                <span
                  class="bp3-file-upload-input"
                >
                  Select a CSV...
                </span>
              </label>
            </div>
            <div>
              <button
                class="bp3-button bp3-intent-primary sc-jTzLTM kWdIFo"
                type="submit"
              >
                <span
                  class="bp3-button-text"
                >
                  Upload File
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
